Src/ILGPU/Runtime/ExchangeBuffer.cs

```diff
@@ -381,4 +458,682 @@ protected override void Dispose(bool disposing)
 
         #endregion
     }
+
+    /// <summary>
+    /// A 2D buffer that stores a specified amount of elements on the associated accelerator
+    /// instance. Furthermore, it keeps a buffer of the same size in pinned CPU memory
+    /// to enable asynchronous memory transfers between the CPU and the GPU.
+    /// </summary>
+    /// <typeparam name="T">The element type.</typeparam>
+    /// <remarks>Members of this class are not thread safe.</remarks>
+    public sealed unsafe class ExchangeBuffer2D<T> : MemoryBuffer, IMemoryBuffer<T>
```

Comment:
Maybe we should come up with an OO design that avoids many code duplications like `CudaViewSource`, which is essentially identical in all classes. This makes maintenance of the code bas significantly easier 🤓

Reply:
Yeah, that's what I was thinking with my suggestion above, before I marked it as ready for review.